common/src/Common/Wallet.hs

```diff
@@ -267,15 +280,21 @@ instance FromJSON AccountGuard where
       keySet = withObject "AccountGuard" $ \o -> do
         keys <- o .: "keys"
         ksPred <- o .: "pred"
-        case mkAccountGuard keys ksPred of
+        ksRef <- o .:? "ref"
```

Comment:
We need to be very careful about these JSON serialization changes.  If they're persisted anywhere and not completely backwards compatible, that will cause problems.  Before merging this we need to understand all the places where this serialization is used and assess the potential impact.